### PR TITLE
#438 use a secret for mutable broker properties

### PR DESCRIPTION
**Pull request by gtully**

### CODE REVIEW

**Comment by brusdev** [comment]
> I was wondering a little if we loose anything with visibility here, but the data the populates the secret is in the CR, so we can always view it there. The secret can protect it when it is deployed

This PR make sense to me because the Kubernetes RBAC is very flexible, so you could have monitor users with no access to the CRDs and secrets data

File: controllers/activemqartemis_controller_test.go [hunk 1]
```diff
@@ -3888,7 +3887,7 @@ var _ = Describe("artemis controller", func() {
 			Expect(createdCrd.Name).Should(Equal(crd.ObjectMeta.Name))
 
 			By("By finding a new config map with broker props")
-			configMap := &corev1.ConfigMap{}
+			configMap := &corev1.Secret{}
```
Comment (brusdev) [inline controllers/activemqartemis_controller_test.go:3890]: could you rename the configMap variables to align them with theirs new type?

File: controllers/activemqartemis_reconciler.go [hunk 2]
```diff
@@ -1668,11 +1671,16 @@ func (reconciler *ActiveMQArtemisReconcilerImpl) NewPodTemplateSpecForCR(customR
...
 
 	reqLogger.Info("Checking out extraMounts", "extra config", customResource.Spec.DeploymentPlan.ExtraMounts)
-	brokerPropertiesConfigMapName := reconciler.addConfigMapForBrokerProperties(customResource)
-	configMapsToCreate := []string{brokerPropertiesConfigMapName}
-	configMapsToCreate = append(configMapsToCreate, customResource.Spec.DeploymentPlan.ExtraMounts.ConfigMaps...)
 
-	extraVolumes, extraVolumeMounts := createExtraConfigmapsAndSecrets(container, configMapsToCreate, customResource.Spec.DeploymentPlan.ExtraMounts.Secrets)
+	configMapsToCreate := customResource.Spec.DeploymentPlan.ExtraMounts.ConfigMaps
+	secretsToCreate := customResource.Spec.DeploymentPlan.ExtraMounts.Secrets
+	resourceName, isSecret := reconciler.addResourceForBrokerProperties(customResource, namer)
```
Comment (brusdev) [inline controllers/activemqartemis_reconciler.go:1677]: why do we need `isSecret`?

File: controllers/activemqartemis_controller_test.go (hunk 1 above)
Comment (gtully) [reply controllers/activemqartemis_controller_test.go:3890]: yep, i'll have that in a later PR, I left them intentionally unchanged to make the diff visible

File: controllers/activemqartemis_reconciler.go (hunk 2 above)
Comment (gtully) [reply controllers/activemqartemis_reconciler.go:1677]: it is for migration, an existing CR can still have a configMap, that will remain till there is some mod, the mount point is then different

**Review by brusdev — APPROVED** [review]

**Pull request merged**